domains/transfer-agreements/TransferAgreements.API/API/Data/TransferAgreementRepository.cs

```diff
@@ -14,7 +14,13 @@ public class TransferAgreementRepository : ITransferAgreementRepository
 
     public async Task<TransferAgreement> AddTransferAgreementToDb(TransferAgreement transferAgreement)
     {
+        var agreements = await context.TransferAgreements.Where(t =>
+            t.SenderId == transferAgreement.SenderId)
+            .ToListAsync();
+        var transferAgreementNumber = agreements.Any() ? agreements.Max(ta => ta.TransferAgreementNumber) + 1 : 0;
```

Comment:
Why is this neccesary now?

Reply:
Did you look into this? 
https://learn.microsoft.com/en-us/ef/core/saving/concurrency?tabs=data-annotations